### PR TITLE
Feature/336 posts research

### LINKED ISSUE #336: @/charterafrica/CMS/Posts
### Description

Implement back-end functionality for various front-end post types. Each of the following can be its own collection:

- [x] News,
- [x] Research,
- [x] Grants,
- [x] Fellowships,
- [x] Events,
- [x] Grantees.

Implementation shouldn't try to implement everything in one PR. Each collection can (and probably should) be implemented in a separate RT. They are included in one issue to guarantee only one person is working on this.

### PR DESCRIPTION
**Pull request by koechkevin**
## Description

Adds Research List collection to CMS.

Fixes #336 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Screenshots

image

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by kelvinkipruto — COMMENTED** [review]
@koechkevin Could you update the description to show what this PR is about?

**Review by kelvinkipruto — APPROVED** [review]

**Review by kilemensi — COMMENTED** [review]
👍🏽 

---

Looking at designs, looks like:

1. Author is only required in Research; optional in News
2. Tags are only required in News; Optional in Research

We need to handle this.

File: apps/charterafrica/src/payload/fields/post.js [hunk 1]
```diff
@@ -9,6 +9,51 @@ const setPublishedOn = ({ value }) => {
   return new Date(value);
 };
 
+export const content = (overrides) => ({
+  name: "content",
+  type: "blocks",
+  blocks: [RichText, MediaBlock, ExternalEmbed],
+  required: true,
+  ...overrides,
```
Comment (kilemensi) [inline apps/charterafrica/src/payload/fields/post.js:17]: 1. Use `@mui/utils` `deepmerge` for overrides (similar to other fields).
2. May be all these fields should be standalone (in own files)?

**Review by kilemensi — COMMENTED** [review]
👍🏽

**Review by kilemensi — COMMENTED** [review]
🚀 

---

My comment was since `content`, `authors`, etc., are reusable, why not have them in own files similar to how we have `link`, `linkArray`, etc.

**Pull request merged**